dimod/include/dimod/quadratic_model.h

```diff
@@ -555,6 +586,73 @@ class BinaryQuadraticModel : public QuadraticModelBase<Bias, Index> {
         }
     }
 
+    /**
+     * Construct a BQM from COO-formated iterators.
+     *
+     * A sparse BQM encoded in [COOrdinate] format is specified by three
+     * arrays of (row, column, value).
+     *
+     * [COOrdinate]: https://w.wiki/n$L
+     *
+     * `row_iterator` must be a random access iterator  pointing to the beginning of the row data.
+     * `col_iterator` must be a random access iterator pointing to the beginning of the column data.
+     * `bias_iterator` must be a random access iterator pointing to the beginning of the bias data.     
+     * `length` must be the number of (row, column, bias) entries.
+     */
+    template <class ItRow, class ItCol, class ItBias>
+    void add_quadratic(ItRow row_iterator, ItCol col_iterator,
+                       ItBias bias_iterator, index_type length) {
+        // determine the number of variables so we can resize ourself if needed
+        if (length > 0) {
+            index_type max_label = std::max(
+                    *std::max_element(row_iterator, row_iterator + length),
+                    *std::max_element(col_iterator, col_iterator + length));
+
+            if ((size_t)max_label >= base_type::num_variables()) {
+                resize(max_label + 1);
+            }
+        } else if (length < 0) {
+            throw std::out_of_range("length must be positive");
+        }
+
+        // count the number of elements to be inserted into each
+        std::vector<size_type> counts(base_type::num_variables());
```

Comment:
Need to give the default value , but it may be zero by default I think.

Reply:
defaults to 0 but I agree we might as well be explicit.